src/auth/guards/magic-link.guard.ts

```diff
@@ -1,8 +1,42 @@
 /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
-import { Injectable } from '@nestjs/common';
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import {
+  BadRequestException,
+  Injectable,
+  NotFoundException,
+  UnauthorizedException,
+} from '@nestjs/common';
 import { AuthGuard } from '@nestjs/passport';
 
 @Injectable()
-export class MagicLinkGuard extends AuthGuard('magic-link') {}
+export class MagicLinkGuard extends AuthGuard('magic-link') {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars, @typescript-eslint/no-explicit-any
+  handleRequest(err: any, user: any, info: any, context: any, status: any) {
+    if (info && !user) {
+      switch (info) {
+        case 'no_account_found':
```

Comment:
In other places in the API (for example [here](https://github.com/iron-fish/ironfish-api/blob/master/src/faucet-transactions/faucet-transactions.service.ts#L55-L58)), we return a `code` and a `message`. It's cleaner to do that if you need to add more error codes in the future, and you can return a single `UnauthorizedException` here.